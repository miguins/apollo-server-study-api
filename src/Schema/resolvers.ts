import { merge } from 'lodash';
import { resolvers as Book } from './Book';
import { resolvers as Product } from './Products';

export const resolvers = merge(
  Product,
  Book
)
