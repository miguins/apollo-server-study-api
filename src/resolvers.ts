import { loadFilesSync } from '@graphql-tools/load-files';
import { mergeResolvers } from '@graphql-tools/merge';
import { productsResolvers } from '@modules/products/graphql/products.resolvers';
import { usersResolvers } from '@modules/users/graphql/users.resolvers';
import path from 'path';

const mergePath = loadFilesSync(
  path.join(__dirname, 'modules/**/graphql/*'), { extensions: ['resolvers.ts'] }
);

const resolvers = mergeResolvers([productsResolvers, usersResolvers]);

export default resolvers;
