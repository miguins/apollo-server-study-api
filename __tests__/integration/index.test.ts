import "reflect-metadata";
import resolvers from '@src/resolvers';
import typeDefs from '@src/schema';
import { ApolloServer } from 'apollo-server-express';

describe('Apollo Server', () => {

  let testServer: ApolloServer;

  beforeAll(() => {
    testServer = new ApolloServer({ typeDefs, resolvers });
  });

  it('should execute query operation', async () => {
    const result = await testServer.executeOperation({
      query: 'query Products { products { id,name,price } }'
    });

    expect(result.errors).toBeUndefined();
    expect(result.data).toBeDefined();
    expect(result).toMatchSnapshot();
  });

  it('should not execute query operation', async () => {

    const result = await testServer.executeOperation({
      query: 'query { products { } }'
    });

    expect(result.errors).toBeDefined();
    expect(result.data).toBeUndefined();
    expect(result).toMatchSnapshot();
  });
});
