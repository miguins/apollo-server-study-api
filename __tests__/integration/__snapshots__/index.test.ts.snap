// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Apollo Server should execute query operation 1`] = `
Object {
  "data": Object {
    "products": Array [
      Object {
        "id": "e9a351a0-0ffe-4027-9e0a-66b44ac88bc5",
        "name": "Product 1",
        "price": 10.5,
      },
      Object {
        "id": "92c22737-5edf-4aef-a045-6988211319c0",
        "name": "Product 2",
        "price": 7.99,
      },
      Object {
        "id": "5b4e3df7-b455-4397-aee4-4b2cfe2e130a",
        "name": "Product 3",
        "price": 80,
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Apollo Server should not execute query operation 1`] = `
Object {
  "data": undefined,
  "errors": Array [
    [SyntaxError: Syntax Error: Expected Name, found "}".],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
